### PR TITLE
Added `-profile wave` to the template.

### PR DESCRIPTION
**Pull request by ewels**
Add a config profile in the template to use Wave.

> [!NOTE]
> Requires `24.03.0-edge`! I considered bumping `nextflowVersion` to `24.04.0` but it feels a bit overkill when the vast majority of end users will not be using `-profile wave`. I'm inclined to mention this in the docs but not encode it in the config. Please shout if you disagree.

Will need more docs about Wave, but that's a larger effort, probably for the website repo.

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] Documentation in `docs` is updated

### CODE REVIEW

**Comment by ewels** [comment]
If someone could add this into upstream pipelines to be sure we're not going to hit any unexpected errors, that'd be fab.

Maybe Sarek @maxulysse @FriederikeHanssen, and rnaseq someone else, at least?

**Review by maxulysse — APPROVED** [review]
<3

**Comment by maxulysse** [comment]
> If someone could add this into upstream pipelines to be sure we're not going to hit any unexpected errors, that'd be fab.
> 
> Maybe Sarek @maxulysse @FriederikeHanssen, and rnaseq someone else, at least?

I'll have a look at adding that in both in POC PR

File: nf_core/pipeline-template/nextflow.config [hunk 1]
```diff
@@ -176,6 +176,11 @@ profiles {
         executor.cpus          = 4
         executor.memory        = 8.GB
     }
+    wave {
+        wave.enabled = true
+        wave.strategy = 'conda,container'
+        wave.freeze = true
+    }
```
Comment (maxulysse) [inline nf_core/pipeline-template/nextflow.config:183]: I would put it before gitpod

**Comment by maxulysse** [comment]
> `nextflow run . -profile test,wave --outdir results -resume`

```
ERROR ~ Error executing process > 'NFCORE_RNASEQ:PREPARE_GENOME:UNTAR_SALMON_INDEX (salmon.tar.gz)'

Caused by:
  Wave invalid response: [400] {"message":"Attribute `buildRepository` must be specified when using freeze mode"}


 -- Check '.nextflow.log' file for details
ERROR ~ Pipeline failed. Please refer to troubleshooting docs: https://nf-co.re/docs/usage/troubleshooting

 -- Check '.nextflow.log' file for details
```

**Review by maxulysse — CHANGES_REQUESTED** [review]
not working as expected

**Comment by maxulysse** [comment]
working locally: https://github.com/nf-core/rnaseq/pull/1291

**Review by maxulysse — APPROVED** [review]
looks good to me

**Review by maxulysse — APPROVED** [review]

File: nf_core/pipeline-template/nextflow.config [hunk 2]
```diff
@@ -171,6 +171,13 @@ profiles {
         shifter.enabled        = false
         charliecloud.enabled   = false
     }
+    wave {
+        apptainer.ociAutoPull   = true
+        singularity.ociAutoPull = true
+        wave.enabled            = true
+        wave.freeze             = true
+        wave.strategy           = 'conda,container'
+    }
```
Comment (mashehu) [inline nf_core/pipeline-template/nextflow.config:180]: please harshil-align

Comment (ewels) [reply nf_core/pipeline-template/nextflow.config:180]: I thought it was? I used @maxulysse's [suggested change](https://github.com/nf-core/tools/pull/2941#discussion_r1584891714) to do so..

Comment (ewels) [reply nf_core/pipeline-template/nextflow.config:180]: I need to coin my own whitespace style.. 🤔 

How about we centre-align on `.` and right align anything on the right hand side of `=`, not including quotes? :trollface: 
```suggestion
    wave {
          apptainer.ociAutoPull =             true
        singularity.ociAutoPull =             true
               wave.enabled     =             true
               wave.freeze      =             true
               wave.strategy    = 'conda,container'
    }
```

Comment (maxulysse) [reply nf_core/pipeline-template/nextflow.config:180]: I'd say it's fine within this scope, we don't have to align all equal signs all over.

@ewels why would you do that?

Comment (mashehu) [reply nf_core/pipeline-template/nextflow.config:180]: all the other equal signs are aligned. consistency! 🙂

Comment (ewels) [reply nf_core/pipeline-template/nextflow.config:180]: The `debug` `nextflow.enable.configProcessNamesValidation` isn't 👀 

Fiiine ye who love massive whitespace diffs, I'll do it..

Comment (ewels) [reply nf_core/pipeline-template/nextflow.config]: Done in e56f5b0d

Comment (mashehu) [reply nf_core/pipeline-template/nextflow.config]: Nice, now only docs are missing

**Review by mashehu — APPROVED** [review]

File: nf_core/pipeline-template/nextflow.config [hunk 3]
```diff
@@ -86,95 +86,102 @@ try {
...
+        singularity.enabled     = false
+        podman.enabled          = false
+        shifter.enabled         = false
+        charliecloud.enabled    = false
+    }
+    wave {
+        apptainer.ociAutoPull   = true
+        singularity.ociAutoPull = true
+        wave.enabled            = true
+        wave.freeze             = true
```
Comment (mirpedrol) [inline nf_core/pipeline-template/nextflow.config:178]: I see in the documentation that we would need to provide `wave.build.repository` together with `wave.freeze = true`. Is it required? Would it be safer to let users provide both parameters.

**Pull request merged**